pkg/server/peer.go

```diff
@@ -491,6 +491,23 @@ func (peer *Peer) handleUpdate(e *FsmMsg) ([]*table.Path, []bgp.RouteFamily, *bg
 					continue
 				}
 			}
+			// RFC4456 8. Avoiding Routing Information Loops
+			// A router that recognizes the ORIGINATOR_ID attribute SHOULD
+			// ignore a route received with its BGP Identifier as the ORIGINATOR_ID.
+			if peer.isIBGPPeer() && !path.IsLocal() {
```

Comment:
peer.isIBGPPeer() accesses to peer.fsm so peer.fsm.lock.RLock() must be called before that.